### PR TITLE
Add exception for when extra parameters are supplied

### PR DESCRIPTION
**Pull request by wilsonge**
This adds custom exception type for when too many parameters are supplied. 

This bit is relatively straightforward so I'm separating it out from the Parameter validation which is probably going to be much more complex

### CODE REVIEW

**Comment by wilsonge** [comment]
OK What's the code sniffer after here? Not seeing it

File: connexion/exceptions.py [hunk 1]
```diff
@@ -95,3 +96,20 @@ def __init__(self, token_scopes, required_scopes, title=None, **kwargs):
         self.missing_scopes = required_scopes - token_scopes
 
         super(OAuthScopeProblem, self).__init__(title=title, **kwargs)
+
+
+class ExtraParameterProblem(ProblemException, BadRequest):
+    def __init__(self, formdata_parameters, query_parameters, title=None, description=None, **kwargs):
```
Comment (rafaelcaricio) [inline connexion/exceptions.py]: We should separate this in two different Exceptions (one for formdata and another for query). The way it is now this exception means "there is a extra parameter in formdata or query or both, I am not sure". This is not good.

Comment (rafaelcaricio) [inline connexion/exceptions.py]: So with this idea we would have a handle of this exception that could render something like:

```json
{
   "type": "https://example.net/validation-error",
   "title": "Your request has extra parameters.",
   "extra-query-params": ["name", "age"],
   "extra-formdata-params": ["reason", "color"]
}
```

Comment (wilsonge) [inline connexion/exceptions.py]: Correct. Or you could have something like:

```
{
    "type": "https://example.net/validation-error",
    "title": "Your request has extra parameters in the query.",
    "extra_params": ["name", "age"],
}
```
Or like you can keep b/c with the existing code (like i have with the description)

The point is having all this information in the exception allows you the choice :) You're giving the user as much information as possible to allow them to render the exception into the format they decide is most optimal for their use case.

Comment (rafaelcaricio) [inline connexion/exceptions.py]: Ok. Sounds good.

**Comment by rafaelcaricio** [comment]
@wilsonge Could you please start some docs about the exceptions in Connexion? If not I will open an issue for that.

**Comment by rafaelcaricio** [comment]
Also please fix the imports ordering:
```patch
--- /home/travis/build/zalando/connexion/connexion/exceptions.py:before	2016-12-14 14:34:03.012488
+++ /home/travis/build/zalando/connexion/connexion/exceptions.py:after	2016-12-14 14:36:53.691374
@@ -1,5 +1,5 @@
-from werkzeug.exceptions import BadRequest, Forbidden, HTTPException,\
-                                Unauthorized
+from werkzeug.exceptions import (BadRequest, Forbidden, HTTPException,
+                                 Unauthorized)
```

**Comment by rafaelcaricio** [comment]
👍

**Comment by wilsonge** [comment]
https://github.com/zalando/connexion/pull/371 Here's a start on the docs

**Comment by hjacobs** [comment]
@wilsonge thanks!

**Pull request merged**